docs/travis.md

```diff
@@ -0,0 +1,102 @@
+# Travis CI
+
+The Zulip server uses [Travis CI](https://travis-ci.org/) for its
+continuous integration.  This page documents useful tools and tips to
+know about when using Travis CI and debugging issues with it.
+
+## Goals
+
+The overall goal of our Travis CI setup is to catch as many possible
+future bugs in Zulip as possible, minimizing delay and false positives
+that waste the time of developers around the Zulip community
+debugging.  There are a few implications of this overall goal:
+
+* If a test is failing nondeterministically in Travis CI, we consider
+  that to be an urgent problem.
+* If the tests become a lot slower, that is also an urgent problem.
+* Everything we do in CI should be to the extent possible runnable in
```

Comment:
Also it took me reading to the end of the sentence and coming back to realize this was meant to describe an alternate environment contrasting with Travis, not to call for things to run fast in Travis by e.g. being unit tests.

Also I think "to the extent possible" can be left out in favor of mentioning the one exception (the Travis config itself) later.

Here's another version:
* Everything we do in CI should also have a way to run it quickly (under 1 minute, preferably under 3 seconds), in order to iterate fast in development. Except when working on the Travis CI configuration itself, a developer should never have to repeatedly wait 10 minutes for a full Travis run.